src/components/auth/ResetPassword.js

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useRef, useState } from 'react';
+import ForgotPasswordStyled from './ForgotPasswordStyled';
+import { useDispatch, useSelector } from 'react-redux';
+import { Formik, Field, Form, ErrorMessage } from 'formik';
+import { message, notification } from 'antd';
+import * as Yup from 'yup';
+import Navbar from '../Navbar';
+import { Link } from 'react-router-dom';
+import { resetPasswordRequestAction } from '../../state/auth/authActionCreator';
+import Spinner from 'react-bootstrap/Spinner';
+import queryString from 'query-string';
+
+function ResetPassword({ history, location }) {
+  const dispatch = useDispatch();
+  const authState = useSelector(state => state.auth);
+  const { loading, error, errResponse, token } = authState;
+  const [passwordDisplay, setpasswordDisplay] = useState({
+    password: null,
+    confirmPassword: null,
+  });
+  const [resetToken, setresetToken] = useState();
+  const [resetEmail, setResetEmail] = useState();
+
+  const passwordRef = useRef();
+  const confirmPasswordRef = useRef();
+  const togglePasswordToText = ref => {
+    const passwordNode = ref.current.childNodes[1];
+    const nodeName = passwordNode.name;
+
+    if (passwordNode.type === 'password') {
+      passwordNode.type = 'text';
+      const tempState = { ...passwordDisplay, [nodeName]: true };
+      setpasswordDisplay(tempState);
+    } else {
+      passwordNode.type = 'password';
+      const tempState = { ...passwordDisplay, [nodeName]: false };
+      setpasswordDisplay(tempState);
+    }
+  };
+
+  const errorClassNames = 'border input border-danger';
+  const validClassNames = 'border input border-green';
+  useEffect(() => {
+    dispatch({ type: 'AUTH_RESET' });
+  }, [dispatch]);
+
+  const openNotification = () => {
+    notification.success({
+      message: 'Password Reset Successful',
+      description: 'You can login into your account with the new password',
+    });
+  };
+
+  useEffect(() => {
+    if (token) {
+      openNotification();
+      //   history.push('/email-verification-sent/');
+    }
+  }, [token, dispatch]);
+  useEffect(() => {
+    if (error) {
+      message.error(errResponse);
+    }
+  }, [error, errResponse]);
+
+  useEffect(() => {
+    const UrlQueryStrings = location.search;
+    const queryValues = queryString.parse(UrlQueryStrings);
+
+    const token = queryValues.token || undefined;
+    const email = queryValues.email || undefined;
+    setResetEmail(email);
+    setresetToken(token);
+
+    // if (!token || !email) {
+    //   history.push('/');
+    // }
+  }, []);
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'location.search'. Either include it or remove the dependency array.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=6093077)